src/Api/SearchQueriesApi.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace LupaSearch\Api;
+
+use LupaSearch\LupaClient;
+
+use function json_encode;
+
+class SearchQueriesApi
+{
+    private $client;
+
+    public function __construct(LupaClient $client)
+    {
+        $this->client = $client;
+    }
+
+    public function getSearchQueries(string $indexId): array
+    {
+        return $this->client->send(
+            LupaClient::METHOD_GET,
+            LupaClient::API_BASE_PATH . "/indices/$indexId/queries",
+            true
+        );
+    }
+
+    public function createSearchQuery(string $indexId, array $httpBody): array
+    {
+        return $this->client->send(
+            LupaClient::METHOD_POST,
+            LupaClient::API_BASE_PATH . "/indices/$indexId/queries",
+            true,
+            json_encode($httpBody)
+        );
+    }
+
+    public function getSearchQuery(
+        string $indexId,
+        string $searchQueryId
+    ): array {
+        return $this->client->send(
+            LupaClient::METHOD_GET,
+            LupaClient::API_BASE_PATH .
+                "/indices/$indexId/queries/$searchQueryId",
+            true
+        );
+    }
+
+    public function updateSearchQuery(
+        string $indexId,
+        string $searchQueryId,
+        array $httpBody
+    ): array {
+        return $this->client->send(
+            LupaClient::METHOD_PUT,
+            LupaClient::API_BASE_PATH .
+                "/indices/$indexId/queries/$searchQueryId",
+            true,
+            json_encode($httpBody)
+        );
+    }
+
+    public function deleteSearchQuery(
+        string $indexId,
+        string $searchQueryId
+    ): array {
+        return $this->client->send(
```

Comment:
I think this method should return `void`.

Reply:
API returns JSON response indicating success/failure message. Due to consistency PHP client returns parsed response body.

Reply:
I disagree with you here, I think the client should look for failure messages and throw exception there. The user should call the delete method and assume this went right. If he has to check - thats a failure from api design point of view.

Reply:
On request failure 4xx, 5xx the client throws ClientException/ServerException

Reply:
Yep, so why do you return stuff to user? On failure it wont return anything.